### PR TITLE
Rodw getting linuxcnc

### PR DESCRIPTION
**Pull request by rodw-au**

### CODE REVIEW

**Comment by rodw-au** [comment]
Seb asked to update the getting linuxcnc docs and this commit does that.
We deleted a lot of old content because a lot of new content had to be added now we are in Debian Bookworm
Users are referred to the 2.8 docs for old content.
A couple of links need updating and the raspberry pi instructions need testing but that is under way.
These places have been marked with FIX_ME

**Review by hansu — COMMENTED** [review]
Some little comments shifted to a separate PR:  https://github.com/rodw-au/linuxcnc/pull/7

File: docs/src/getting-started/getting-linuxcnc.adoc [hunk 1]
```diff
@@ -23,158 +23,56 @@ without making any permanent changes to your computer) or booting the
...
-.zsync in Windows
+Another alternative for the Raspberry Pi is to download the Debian Bookworm image from https://raspi.debian.net/tested-images/
 
-There is a Windows port of zsync. It works as a console application. It
-can be downloaded from:
-
-https://www.assembla.com/spaces/zsync-windows/documents
+This Pi image can  be installed using the normal Pi
+https://www.raspberrypi.org/documentation/installation/installing-images/README.md[install process]
+and using the Raspberry Pi Imager app. Note the Debian image will require additional configuration as by default it does not install a graphical environment and only presents a text based console. How to complete the installation will be covered in a later section of this document.
```
[the remark this answers is not in the document]
Comment (rodw-au) [reply docs/src/getting-started/getting-linuxcnc.adoc:55]: Thanks. Maybe review it when the Pi picks up Bookworm.

Do we really want to use a non Debian Image?
As the document is written configuring the tweaks are identical on x86 and the PI following installation
Further on, the user only has to type one line to get a desktop so its not a massive additional process
`apt install task-xfce-desktop`

[the remark this answers is not in the document]
Comment (hansu) [reply docs/src/getting-started/getting-linuxcnc.adoc:55]: >Do we really want to use a non Debian Image?

You mean an image that is not officially provided from the Debian website?
I don't know what desktop they are using, but if that works well and if its easier for the users, why not?

[the remark this answers is not in the document]
Comment (SebKuzminsky) [reply docs/src/getting-started/getting-linuxcnc.adoc:55]: There are three options for a Raspberry Pi image for LinuxCNC (Debian, Raspberry Pi OS, and Ubuntu), all of them unfortunately suboptimal. There's some discussion on the topic here, but no consensus yet: https://github.com/LinuxCNC/linuxcnc/issues/2349

**Comment by SebKuzminsky** [comment]
Thanks for working on this super important document.

This is (hopefully) one of the first docs that new users read when they get serious about trying LinuxCNC.  As such, it's important that it be concise and simple to follow.  IMO we've historically done a bad job of that...

I'd ideally like to see a super simple recipe that will work for almost everyone, something like:
1. Download *this* Live/Install ISO.
2. Do *this* to boot the Installer, and here's some help on setting up your system.
3. Do *this* to check latency in an installed system.  (Note that it's tricky to test latency in a Live system, because you'd have to know the CPU count and edit the kernel command-line arguments in GRUB, which IMO would be a distraction in a "Getting Started" document.)
4. Do *this* to start some specific LinuxCNC sim config, to see it run.

This simple and reliable recipe should be highly visible right up at the top.  Anything about customizing, optimizing, troubleshooting, and alternate install methods should come later, where it won't distract and put off new users.

**Comment by hansu** [comment]
>This simple and reliable recipe should be highly visible right up at the top. Anything about customizing, optimizing, troubleshooting, and alternate install methods should come later, where it won't distract and put off new users.

I moved it a bit up in [my suggestion](https://github.com/hansu/linuxcnc/blob/getting-linuxcnc-comments/docs/src/getting-started/getting-linuxcnc.adoc)

**Comment by rodw-au** [comment]
The question I have is will Linuxcnc offer prebuilt ISOs with 2.9 or will the Debian Bookworm ISO and using the repositories  the only way to get Linuxcnc? If so will we have it by 10 June? 

Catering for both really confuses the instructions. Please make a decision and advise.

If the release of a .ISO image is not imminent, It references to it should be deleted for now. V 2.9 users need this information now.

Also note it's not possible to provide a definitive download link until after the Debian Bookworm is released on 10 June. Eg this was written when RC2 was the best option, and now RC3  has been released.

FYI I also have a project running with the Kernel RT team to try and get rid of all the mess with Realtek drivers.

Is there a method to squash my commits into one?

File: docs/src/getting-started/getting-linuxcnc.adoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (rodw-au) [reply docs/src/getting-started/getting-linuxcnc.adoc:55]: Everything about the Pi is suboptimal. The only way forward is to follow David Smith's method. I added a git link to #2349 
eg.  build the PREEMPT_RT kernel in the Raspberry crosss compiler build environment. 
This gives the user a simple image to install in the usual way.

[the remark this answers is not in the document]
Comment (rodw-au) [reply docs/src/getting-started/getting-linuxcnc.adoc:55]: The method we proposed using the Bookworm repos generates an acceptable result despite being long winded. It is structured to minimize the mount of time the user has to use the CLI.

**Comment by rodw-au** [comment]
> > Is there a method to squash my commits into one?
> 
> Sure. There are lots of HowTos for that, for example this one: https://www.internalpointers.com/post/squash-commits-into-one-git

Thanks Hans, I was able to squash my commits on my local copy last night but could not push it to my fork. I get this error:
```
debian@debian:~/devel/linuxcnc-rod$ git push
Username :        
Password: 
To https://github.com/rodw-au/linuxcnc
 ! [rejected]              rodw-getting-linuxcnc -> rodw-getting-linuxcnc (non-fast-forward)
error: failed to push some refs to 'https://github.com/rodw-au/linuxcnc'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
```
Any ideas on how get past this?

**Comment by hansu** [comment]
With a force push: `git push -f`

**Comment by rodw-au** [comment]
Could you guys please review this version published on my home NAS and provide feedback here.
[http://110.238.151.153/linuxcnc/getting-linuxcnc.html](url)

Git does not lend itself well to displaying asciidoc.

We thought showing how to install linuxcnc on a PC that already has Debian Bookworm installed simplified the process.
Installing Debian Bookworm on x86 and the raspberry pi are presented as separate sections.

**Comment by hansu** [comment]
The PC section looks good to me. Cannot say much to the Raspi section as I haven't use LinuxCNC on that platform.

**Comment by rodw-au** [comment]
> The PC section looks good to me. Cannot say much to the Raspi section as I haven't use LinuxCNC on that platform.

Thanks, I'll check it again and update GIT in the next day or so. I think it's best you stay ignorant of the Pi it can be  a very frustrating device :) 

I think today there are better ARM based PC candidates but Installation is not trivial as you can't simply use Debian's RT image...

**Comment by rodw-au** [comment]
Debian 12 Bookworm is now released!
They have a few things to sort out on their website yet, but you can choose the XFCE ISO from this page
https://cdimage.debian.org/debian-cd/current-live/amd64/iso-hybrid/

The problem is the version of 2.9 in Debian 12 is quite old but now Seb has a Buildbot here
http://buildbot2.highlab.com/

How can we set up our sources.list to look in this repo first before the official repo? Can someone help here so I can include the changes in this document?

Alternatively, could a later version of Linuxcnc 2.9 be sent to Debian Unstable now? That might be easier to set APT priorities but it won't stay current.

**Comment by rodw-au** [comment]
This is now a final version following the release of Debian 12 Bookworm which only stabilised yesterday.
You can preview it as HTML here:
http://vmnet.com.au/linuxcnc/getting-linuxcnc.html

We now offer the user a choice of using the smaller netinst .ISO (preferred) or the complete XFCE live install where no network exists on the target PC.

We also include instructions on how to override the default Debian repositories in favour of Seb's secret buildbot by pinning the linuxcnc files there to our sources.list.

If someone could push this so it gets into the online docs for 2.9 it would be fantastic as the info here is much needed by our users.

I was thinking that when a new version of linuxcnc V 2.9 is pushed to Debian unstable/testing, we could update the apt pinning  to get the files from there.

I might add that the pinning works for master branch (V2.10) from the buildbot but I restricted it to V 2.9 only.

**Comment by rodw-au** [comment]
Closing. Replaced by #2522

**Pull request closed without merging**